app/controllers/APIController.scala

```diff
@@ -34,6 +34,23 @@ class APIController @Inject() (
       }
   }
 
+  def createProduct() = Action.async { implicit request: Request[AnyContent] =>
+    val reqJson = request.body.asJson
```

Comment:
Notice that you are believing the user will give you good values here. what happens if they send an empty string or an empty json? A way to avoid this is to validate input in the controller before sending it further downstream to the logic. A simple way to do this is to declare a "request" case class, or request model to read your input to, and then pass that request instance down to your database logic.